sirepo/template/srw_importer.py

```diff
@@ -461,9 +459,11 @@ def _get_default_drift():
     Returns:
         str: default drift propagation paramters
     """
-    c = pkio.read_text(_JS_DIR.join('srw.js'))
-    m = re.search(r'function defaultDriftPropagationParams.*?return\s*(\[[^\]]+\])', c, re.DOTALL)
-    return pkjson.load_any(m.group(1))
+    return pkjson.load_any(re.search(
```

Comment:
This is an example where naming the function, e.g. re.search, is a good idea. Even if it is a nested function, it avoids too much nesting without naming. Using intermediate variables is a way of naming, but it is better to name the function begin performed as a functional call.

Reply:
I'm having a hard time understanding what you're saying. Can you say it in a different way? 

Should I make a new named inner function for just the re.search bit? Should I revert back to the original structure so `m` names the re.search? 

When you say "nested function" do you mean an inner function (ex a new `def ...` inside of '`def _get_default_drift` or do you mean that my refactoring where I nested the function inside of another function and removed the local var?

Reply:
[Nested Function](https://en.wikipedia.org/wiki/Nested_function) is the generic, "computer science" term for Python's inner functions.

Yes, make an inner function or perhaps not a nested function. 

This is a bit like the [Rule of Three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)). Some people say that when a function is longer than a screen, it needs to be broken up. Others say more than 15 lines. Yet others (and me) says when you can name the function, name it as a function, not as a temporary variable, which creates problems. A function and a variable are layers of indirection, but the function is stateless, and the variable is stateful, so it's the loser in the naming contest. If a function can be stateless, better to make it so.

Reply:
Thanks:) c61058932

Is there a name for a function called from within another function? Ex what do you call the function bar in `foo(bar())`

Reply:
Just "call" afaik. If `bar()` is called within `bar()` then it is called a recursive call. If `bar()` is called within an outer function it is called a nested call. The name is based on how the callee is defined.